src/main/scala/io/iohk/ethereum/mining/BlockGenerator.scala

```diff
@@ -37,11 +37,13 @@ class BlockGenerator(blockchainStorages: BlockchainStorages, blockchainConfig: B
         val blockTimestamp = blockTimestampProvider.getEpochSecond
         val header: BlockHeader = prepareHeader(blockNumber, ommers, beneficiary, parent, blockTimestamp)
 
-        val transactionsForBlock = transactions
-          .filter(_.tx.gasLimit < header.gasLimit)
-          //if we have 2 transactions from same address we want first one with lower nonce
-          .sortBy(_.tx.gasPrice)
-          .sortBy(_.tx.nonce)
+        val sortedTransactions = transactions.groupBy(_.senderAddress).values.toList.flatMap { txsFromSender =>
+          val ordered = txsFromSender.sortBy(_.tx.gasPrice).reverse.sortBy(_.tx.nonce)
```

Comment:
Could we maybe use only one tx per sender and nonce? This is for example when we receive 2 txs from a peer with the same sender and nonce (if someone is overriding one of it's txs). I think that in `Ledger.prepare` we already remove txs whose execution fails so this is up to discussion.